script.js

```diff
@@ -1,12 +1,91 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+window.onload = setTimeout(setup, 500);
+
+setTimeout(rootSetup, 500);
+
+// function getAllShows() {
+//   return fetch("https://api.tvmaze.com/shows")
+//     .then((response) => response.json())
+//     .then((data) => {
+//         data.forEach(({name, id}) => {
+//         availableShows.push({name, id});
+//       });
+//     })
+//     .catch((error) => console.log(error));
+// }
+
+// pull all episodes from the selected show
+
+function createShowList() {
+  let showList = [];
+  availableShows.forEach((show) => showList.push(show.name));
+  let select = document.querySelector("#show-selector");
+  showList.sort().forEach((show) => {
+    let option = document.createElement("option");
+    option.innerText = show;
+    select.appendChild(option);
+  });
 }
+setTimeout(createShowList, 300);
+
+// show chosen show
+function showChosen (event) {
+  
+  let chosenShow = availableShows.filter(
+    (show) => show.name === event.target.innerHTML ||
+      show.name === event.target.value
+  );
+  
+  showIdNumber = chosenShow[0].id;
+
+  allEpisodes = [];
+  getData().then((episode) => {
+    episode.forEach((element) => allEpisodes.push(element));
+  });
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+  episodesDiv.innerHTML = "";
+  setTimeout(rootSetup, 500);
+  document.querySelector("#episode-selector").innerHTML = "";
+  setTimeout(optionList, 1000);
 }
+document.querySelector("#show-selector").addEventListener("change", (event) => {
+  showChosen(event);
+  // let chosenShow = availableShows.filter(
```

Comment:
I'd recommend deleting the commented out code - it adds noise to the code you have (makes it harder to see what code you actually have that you're using) :)